WcaOnRails/app/models/registration.rb

```diff
@@ -102,7 +107,7 @@ def waiting_list_info
 
   validate :must_register_for_gte_one_event
   private def must_register_for_gte_one_event
-    if events.empty?
+    if registration_events.length { |r| !r.marked_for_destruction? } == 0
```

Comment:
@viroulep we use `accept_nested_attributes_for` and we want to do this validation before they are saved. Note that we can remove events this way (by setting `_destroy: '1'` in the attributes for some events) and `self.events` loads the events via registration_events from db and it doesn't care if currently some are *marked for destruction* as they hasn't been saved yet.